server/api.js

```diff
@@ -4,15 +4,140 @@ import pool from "./db";
 
 const router = Router();
 
+// Get all the tasks
 router.get("/tasks", (req, res) => {
 	pool.query("SELECT * FROM tasks").then((result) => {
-		// console.log(result.rows);
 		return res.send(result.rows);
 	});
 });
-
 router.get("/", (_, res) => {
-	return res.json({ message: "Hello, from the api route!" });
+	res.json({ message: "Hello, from the api route!" });
 });
 
+router.post("/tasks", (req, res) => {
+	const { task_title, due_date, evidence, status_title, element_title } =
+		req.body;
+	let params = [];
+	if (!task_title) {
+		return res
+			.status(404)
+			.send({ success: false, message: "Task title cannot be empty" });
+	} else {
+		params.push(task_title);
+	}
+	if (due_date.length > 0) {
+		params.push(due_date);
+	} else {
+		params.push("");
+	}
+	if (evidence.length > 0) {
+		params.push(evidence);
+	} else {
+		params.push("");
+	}
+	if (status_title.length > 0) {
+		pool
+			.query("SELECT Status_id FROM status WHERE Status_title = $1", [
+				status_title,
+			])
+			.then((result) => {
+				params.push(result.rows[0].status_id);
+			})
+			.catch((err) => console.log(err));
+	} else {
+		res
+			.status(404)
+			.send({ success: false, message: "status_title cannot be empty" });
+	}
+	if (element_title.length > 0) {
+		pool
+			.query("SELECT Element_id FROM elements WHERE Element_title = $1", [
+				element_title,
+			])
+			.then((result) => {
+				params.push(result.rows[0].element_id);
+			})
+			.catch((err) => console.log(err));
+	} else {
+		res
+			.status(404)
+			.send({ success: false, message: "element_title cannot be empty" });
+	}
+
+	pool
```

Comment:
insert a check here we need to make sure that if the user did not choose to send all details what should happen if null? research in POSTGRES COALESCE, its very useful in this case to avoid error and not overwrite in the future as it takes several arguments

https://www.postgresqltutorial.com/postgresql-coalesce/

Reply:
I though That the user can not submit if this is null, restricted from the front end .

Reply:
In general: don't rely on the client (the React app, in this case) validate the data for you! It's entirely possible to make a request to the Express server from another client, e.g. Postman or just writing your own `fetch` in the browser console. In some cases that's OK, because the request will just be invalid and therefore basically ignored, but if it could lead to e.g. saving corrupt data into the DB or crashing the app it needs to be checked.

Reply:
Thank you , I haven't thought of this ! good to know .